### ISSUE TITLE
A bug in the undo operation

### ISSUE THREAD

**Issue by sirius-fan** [NvChad/NvChad#1384]
**Describe the bug**
First, 
I have set not to record history file modifications in my configuration file.
 `vim.opt.undofile = false`
Second,
When I open any file and then do undo directly, it shows that a change to the last line was undone.

**To Reproduce**
1. open a file
2. press ‘u’

**Screenshots**

https://user-images.githubusercontent.com/25720015/181228831-692303e1-352b-4c6e-933c-5db50d8b2226.mp4

**Desktop (please complete the following information):**
 - Operating System 
 manjaro linux
 - Terminal 
 konsole
 - Version of Neovim    
 NVIM v0.7.2

**Additional context**
 `PackerSync` has been executed.
When removing the personal configuration(custom folder), the problem persists.

**Comment by tomasky** [comment]
try this:

```lua
--custom/init.lua
vim.opt.undofile = false
```

**Comment by sirius-fan** [comment]
> try this:
> 
> ```lua
> --custom/init.lua
> vim.opt.undofile = false
> ```

I'm sure I've set

>Describe the bug
>First,
>I have set not to record history file modifications in my configuration file.
>vim.opt.undofile = false

and do `:echo &undofile` ， the output is 0

**Comment by tomasky** [comment]
add this in file `custom/init.lua`

**Comment by tomasky** [comment]
@siduck  maybe you should start adding changelogs now

**Comment by sirius-fan** [comment]
> add this in file `custom/init.
![2022-07-27-191907_1920x1080_scrot](https://user-images.githubusercontent.com/25720015/181234978-2fa56db3-db2a-453d-9cb3-c21e4b084f19.png)
lua`

**Comment by sirius-fan** [comment]
I really added, do I need to record a video?

**Comment by siduck** [comment]
> I really added, do I need to record a video?

yes do it

**Comment by siduck** [comment]
> @siduck maybe you should start adding changelogs now

https://github.com/NvChad/NvChad/issues/1385

**Comment by tomasky** [comment]
i cant reproduce it

**Comment by siduck** [comment]
same

**Comment by siduck** [comment]
Uploading simplescreenrecorder-2022-07-27_17.02.31.mp4…
@sirius-fan

**Comment by tomasky** [comment]
@sirius-fan  , try to remove vim.cmd line ,

**Comment by tomasky** [comment]
@sirius-fan  , try to remove vim.cmd line ,

**Comment by sirius-fan** [comment]
https://user-images.githubusercontent.com/25720015/181238362-e5214cc7-c1b6-4c5d-960c-48464d70333e.mp4

@siduck

**Comment by tomasky** [comment]
try this:
```lua
vim.opt.undofile = false
vim.cmd[[:command -nargs=O Q :qall]]

```

**Comment by sirius-fan** [comment]
> try this:
> 
> ```lua
> vim.opt.undofile = false
> vim.cmd[[:command -nargs=O Q :qall]]
> ```
useless

I upload my config folder

[nvim.tar.gz](https://github.com/NvChad/NvChad/files/9198303/nvim.tar.gz)


And tested without using the terminal, the nevide also has problems, downloaded nvim.appimage from the official repo releases, there are still problems.

**Comment by tomasky** [comment]
i cant reproduct using your config

**Comment by sirius-fan** [comment]
```
# sirius @ fan-PC in ~ [20:45:22] 
$ rm -rf ~/.config/nvim
rm -rf ~/.local/share/nvim
rm -rf ~/.cache/nvim

# sirius @ fan-PC in ~ [20:45:28] 
$ git clone https://github.com/NvChad/NvChad ~/.config/nvim --depth 1 && nvim

正克隆到 '/home/sirius/.config/nvim'...
remote: Enumerating objects: 44, done.
remote: Counting objects: 100% (44/44), done.
remote: Compressing objects: 100% (42/42), done.
remote: Total 44 (delta 0), reused 25 (delta 0), pack-reused 0
接收对象中: 100% (44/44), 38.12 KiB | 609.00 KiB/s, 完成.

# sirius @ fan-PC in ~ [20:46:41] 
$ nvim newfile

```
After I reinstalled, the problem still exists and can be reproduced on my Windows and ubuntu server

**Comment by tomasky** [comment]
where is your chadrc.lua file ?

**Comment by tomasky** [comment]
@sirius-fan 

```bash
cp -r examples lua/custom
```

**Comment by sirius-fan** [comment]
I deleted all my configuration, in order to exclude the influence of my personal configuration @tomasky 

`cp -r examples lua/custom` 
I don't think it makes sense to copy two almost empty files，

**Comment by tomasky** [comment]
empty files ? ,no , chardrc.lua is not empty .

**Comment by sirius-fan** [comment]
@tomasky 

yep, it sets a theme, so almost empty，copying doesn't solve the problem
Although I don't think chardrc.lua should be a must，
This file should not be required before
chadrc_exists in the code
```lua
local chadrc_exists, chadrc = pcall(require, "custom.chadrc")
```
but your pr today doesn't judge whether it exists or not（in remove_default_keys）
So now it is necessary
My fault

**Comment by siduck** [comment]
@siduck chadrc is not a must 

utils.lua

![image](https://user-images.githubusercontent.com/59060246/181266292-71da1977-b8e9-4213-9ca6-ec178c59f714.png)

**Comment by sirius-fan** [comment]
I rolled back the git repo using git-reset
and The first version with the problem was https://github.com/NvChad/NvChad/commit/a83ebc34e547c0e28022e2a5f414deaba22c485f
The previous version https://github.com/NvChad/NvChad/commit/5d4c51127c4baffa37757de9f5ec816adf4eb690 did not have this problem
![image](https://user-images.githubusercontent.com/25720015/181271716-436362fc-aa43-4ad2-8176-dbc081e32a40.png)
I'm not in a hurry, wait for others to reproduce

**Comment by siduck** [comment]
@sirius-fan i cant reproduce this issue at all, can you provide steps to reproduce :/

**Comment by tomasky** [comment]
if set `vim.opt.undofile = false`
and do :echo &undofile ， the output is 0  , the result maybe correct

if i set `vim.opt.undofile = true`, the output is 1

**Comment by sirius-fan** [comment]
I forgot to open Screenkey
new video shows rollback operation
Other steps have been mentioned earlier
**TLDR:** `nivm newfile` -> press u -> `:q` -> can not exit -> Ctrl+r -> `:q` -> succeed exit 

If you can't reproduce it, there's nothing I can do

Chinese in the video:
已位于最旧的改变->Already at oldest change 
已位于最新的改变->Already at newest change 

https://user-images.githubusercontent.com/25720015/181282395-0bc982f0-d9bb-425a-ade9-ff4ad9ae2102.mp4

**Comment by tomasky** [comment]
oh ,i reproduce it

**Comment by tomasky** [comment]
@siduck  it is related to these lines:
```lua
--lazy_load.lua

if tb.plugin == "nvim-lspconfig" then
              vim.cmd "silent! e %"
end
```

**Comment by siduck** [comment]
@tomasky how is that related to this :thinking:

**Comment by tomasky** [comment]
i dont know , but if i remove `vim.cmd "silent! e %"` , the problem gone

**Comment by tomasky** [comment]
Besides,  @siduck , Almost every file  will load  lspconfig plugin,
i think you should can adapt the filetype
 
```lua
M.on_file_open = function(plugin_name)
  M.lazy_load {
    events = { "BufRead", "BufWinEnter", "BufNewFile" },
    augroup_name = "BeLazyOnFileOpen" .. plugin_name,
    plugin = plugin_name,
    condition = function()
      local file = vim.fn.expand "%"
      return file ~= "NvimTree_1" and file ~= "[packer]" and file ~= ""
    end,
  }
end
```

**Comment by tomasky** [comment]
Besides,  @siduck , Almost every file  will load  lspconfig plugin,
i think you should adapt the filetype

 
```lua
M.on_file_open = function(plugin_name)
  M.lazy_load {
    events = { "BufRead", "BufWinEnter", "BufNewFile" },
    augroup_name = "BeLazyOnFileOpen" .. plugin_name,
    plugin = plugin_name,
    condition = function()
      local file = vim.fn.expand "%"
      return file ~= "NvimTree_1" and file ~= "[packer]" and file ~= ""
    end,
  }
end
```

**Comment by tomasky** [comment]
Besides,  @siduck , Almost every file  will load  lspconfig plugin,
i think you should adapt the filetype

 
```lua
M.on_file_open = function(plugin_name)
  M.lazy_load {
    events = { "BufRead", "BufWinEnter", "BufNewFile" },
    augroup_name = "BeLazyOnFileOpen" .. plugin_name,
    plugin = plugin_name,
    condition = function()
      local file = vim.fn.expand "%"
      return file ~= "NvimTree_1" and file ~= "[packer]" and file ~= ""
    end,
  }
end
```

**Comment by tomasky** [comment]
@siduck  these lines are  not needed
```lua
if tb.plugin == "nvim-lspconfig" then
              vim.cmd "silent! e %"
end
```

**Comment by sirius-fan** [comment]
In fact, it is triggered by the `:e` command

When just doing `:e` , not `:e filename`
creat a new file , input  `:e` + Enter.

It appears in both nvim and vim. I don't know if it's a feature or a bug.
In short, it can be avoided by not executing the empty `:e` command

@tomasky @siduck

**Comment by tomasky** [comment]
no,  `:e filename` or `:e %`   or `:e`, them should be the same

**Comment by siduck** [comment]
> @siduck these lines are not needed
> 
> ```lua
> if tb.plugin == "nvim-lspconfig" then
>               vim.cmd "silent! e %"
> end
> ```

lspconfig will not start unless in insert mode then, thats why I added it

**Comment by tomasky** [comment]
> > @siduck these lines are not needed
> > ```lua
> > if tb.plugin == "nvim-lspconfig" then
> >               vim.cmd "silent! e %"
> > end
> > ```
> 
> lspconfig will not start unless in insert mode then, thats why I added it

i cant reproduct that ,if i remove these lines ,my lspconfig plugin loads and  works normally

**Comment by tomasky** [comment]
but my lsp key mappings not be loaded
